src/scheduler/node_info.c

```diff
@@ -3214,7 +3210,7 @@ is_vnode_eligible(node_info *node, resource_resv *resresv,
 	}
 
 	/* Does this chunk have AOE? */
-	if (resresv->aoename != NULL) {
+	if ((resresv->aoename != NULL && pl->pack)) {
```

Comment:
Why are you checking for pack?  Won't this break the default case of place=free?

Reply:
We don't want to mark non-provisionable nodes ineligible in case of place=free, Since it might get used to satisfy a non-provisionable sub-chunk.

Reply:
I still don't understand.  From what I see, you are only going to do provisioning for place=pack.  You already handle the case of provisioning chunk vs non-provisioning chunk with resresv->is_prov_needed.

Reply:
It might be possible that I didn't understand it well, so please correct if I am wrong. This check is in is_vnode_eligible() which is called from check_node_array_eligibility(), as I understand this function, marks node eligible and in-eligible for job or reservation request. As part of this RFE, we are allowing user to request aoe just like any other resource. Earlier requesting aoe in one sub-chunk makes it job wide, this behaviour was done using resresv->aoename variable. In is_vnode_eligible() earlier only those nodes were marked eligible which had the requested aoe( since it was job wide), but now we want other nodes to be also marked eligible, since aoe as per this RFE will no more be a job-wide applicable resource, and will be satisfied on per chunk basis.

So this check is valid only when place=pack, when user want all the sub-chunks to be satisfied by the same node. Now I feel that we can remove this check itself, since AOE should behave like any other resource request. Let me know your thoughts on removing the aoe check for node eligibility.

Reply:
is_vnode_eligible() is done for the whole select, not per-chunk.  There is a different is_vnode_eligible() function for chunks.  Since nodes may be eligible for some chunks and not others depending on aoe, this check doesn't belong here anymore.  If it is needed to check per-chunk, then look at is_vnode_eligible_chunk().

Reply:
Done. Thanks